### PR TITLE
Fix FPEs in tests

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

There are a couple MC neutrino tests I commented out because they're too hard for me to fix. That code isn't currently being used in simulations and will need a lot more debugging and testing in general. I'll talk with Francois Foucart about getting that fixed separately. My main goal is to just get FPEs on again in tests.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/NumericalAlgorithms/RootFinding/TOMS748.hpp [hunk 1]
```diff
@@ -345,10 +360,14 @@ std::pair<T, T> toms748_solve(F f, const T& ax, const T& bx, const T& fax,
   T e(static_cast<T>(1e5F));
   T fd(static_cast<T>(1e5F));
 
-  T c(std::numeric_limits<T>::signaling_NaN());
-  T d(std::numeric_limits<T>::signaling_NaN());
+  // T c(std::numeric_limits<T>::quiet_NaN());
+  // T d(std::numeric_limits<T>::quiet_NaN());
```
Comment (wthrowe) [inline src/NumericalAlgorithms/RootFinding/TOMS748.hpp]: Remove.

File: src/Evolution/Systems/GrMhd/ValenciaDivClean/FixConservatives.cpp [hunk 2]
```diff
@@ -350,13 +351,21 @@ bool FixConservatives::operator()(
                                          : static_cast<SimdType>(0);
 
     // Equation B.27 of Foucart
+    //
+    // We avoid division by zero in the SIMD case using a mask and effectively
+    // set the result to zero by dividing by the max double.
```
Comment (wthrowe) [inline src/Evolution/Systems/GrMhd/ValenciaDivClean/FixConservatives.cpp]: It looks like you decided on a different implementation than this, without the max double part.

File: src/Time/ChooseLtsStepSize.cpp [hunk 3]
```diff
@@ -22,7 +22,10 @@ TimeDelta choose_lts_step_size(const Time& time, const double desired_step) {
   // n because of floating-point truncation.  The inner ceil call
   // avoids this problem.
   const auto desired_step_power =
-      static_cast<size_t>(std::ceil(std::log2(std::ceil(desired_step_count))));
+      desired_step == std::numeric_limits<double>::infinity()
```
Comment (wthrowe) [inline src/Time/ChooseLtsStepSize.cpp]: This needs an `abs()`, since the step can be negative, or check for `desired_step_count == 0.0` instead.

**Review by wthrowe — CHANGES_REQUESTED** [review]
Thanks for going through all these.  I agree we can deal with the MC problems later.

**Comment by nilsdeppe** [comment]
I made the changes. Since they are pretty small I just went ahead and squashed it. Partly to get this in before more FPEs appear 🙃

**Review by wthrowe — APPROVED** [review]
Looks good.  I would have requested a direct squash if I'd thought of it.

**Pull request merged**